64bits update to tango.core.Traits and tango.util.container.Slink

bitcuits opened the pull request:
The function staticArraySize now calculates the size of the array from the size of the Element in the array instead of the (_T.ptr) which sometimes seem to be the size of void_ in 64bits.

 And the function Slink.nth(int n) is change to Slink.nth(size_t n) which more compatible with 32/64bits system.

Commit:
ConstOf HolderOf MutableOf added to Traits.d

Commit:
Merge remote-tracking branch 'upstream/d2port' into d2port

Commit:
real is 0 bug !\?fixed

Commit:
Merge remote-tracking branch 'siegelord/d2port' into d2port

Commit:
In tango.util.container Slink nth(int n) change to nth(size_t n) to support 64bit

Commit:
False commit reverted

Commit:
False commit reverted

Commit:
Merge branch 'd2port' of https://github.com/SiegeLord/Tango-D2 into d2port

SiegeLord commented:
Err... could you make a new branch off of d2port branch with just the commits you want merged?

Commit:
Merge branch 'siegelord' into cbleser

Commit:
cbleser/d2port is now the same as siegelord/d2port

Commit:
tango/util/container/Slink.d
findKey and nth function change to support all mutual types

Commit:
tango/util/container/Container.d  int size parameter change to size_t to support 64bit coding better

Commit:
size function in tango/util/container/RedBlack.d is now const again

Commit:
size function in tango/util/container/HashSet.d is now const

Commit:
size function in tango/util/container/SortedMap.d is now const

Commit:
tango/util/container/more/BitSet.d
dup function changed to inout type
and size function changed to const

Commit:
size function in tango/util/container/more/Vector.d is now const

Commit:
size function in tango/util/container/more/Stack.d is now const

Commit:
size function in tango/util/container/more/CacheMap.d is now const

Commit:
tango/util/container/more/StackMap.d
StackMap is dynamically resizable instead of fixed size

bitcuits commented:
I have added some small changes to tango.util.container.
Some member function has checked from mutable to const or inout, this makes it more convenient to use in const functions.
StackMap can now grow. Which also makes it more usable.

Commit:
size function in tango/util/container/HashMap.d is now const

Commit:
tango/util/container/LinkedList.d
Some member function has change to const and LinkedList now support.
opAssignIndex and opIndex
